src/wp-includes/html-api/class-wp-html-processor-state.php

```diff
@@ -140,4 +248,169 @@ public function __construct() {
 		$this->stack_of_open_elements     = new WP_HTML_Open_Elements();
 		$this->active_formatting_elements = new WP_HTML_Active_Formatting_Elements();
 	}
+
+	/**
+	 * Runs the reset the insertion mode appropriately algorithm.
+	 *
+	 * @since 6.7.0
+	 *
+	 * @throws WP_HTML_Unsupported_Exception When encoutering unsupported nodes.
+	 *
+	 * @see https://html.spec.whatwg.org/multipage/parsing.html#reset-the-insertion-mode-appropriately
+	 */
+	public function reset_insertion_mode(): void {
+		/*
+		 * > 1. Let _last_ be false.
+		 * > 2. Let _node_ be the last node in the stack of open elements.
+		 * > 3. _Loop_: If _node_ is the first node in the stack of open elements, then set _last_
+		 * >            to true, and, if the parser was created as part of the HTML fragment parsing
+		 * >            algorithm (fragment case), set node to the context element passed to
+		 * >            that algorithm.
+		 * > …
+		 */
+		$last       = false;
+		$last_index = $this->stack_of_open_elements->count() - 1;
```

Comment:
earmarking this for when we support `'marker'` which might be _very soon_

at that point we have to decide if `count()` returns the count _with_ markers or without. also we need to confirm with the spec which that should be here